### PR TITLE
Adjust status enums

### PR DESCRIPTION
**Pull request by AlyssaCote**
This PR addresses a status bug that occurred when Experiment Telemetry is disabled. If it's disabled, all statuses would read `Pending` even though those statuses would never show up. Because the dashboard can't know for sure whether the statuses will eventually come through, I thought that `Unknown` is a better fit for that state. If the dashboard can't read from the status directory (because of a `KeyError` or `JSONDecodeError`), the status read out as `Unknown`. Now that `Unknown` is being used in place of what used to be `Pending`, I added a new enum called `Malformed` that handles statuses in that state.

in short,
`UNKNOWN` is now `MALFORMED`
`PENDING` is now `UNKNOWN`

### CODE REVIEW

File: smartdashboard/utils/StatusReader.py [hunk 1]
```diff
@@ -177,22 +177,26 @@ def get_experiment_status_summary(runs: t.Optional[t.List[Run]]) -> str:
         orcs = [orch for run in runs for orch in run.orchestrator]
         shards = [shard for orc in orcs for shard in orc.shards]
 
+        unknown_counter = 0
         for entity in itertools.chain(apps, ens_members, shards):
-            unknown_counter = 0
```
Comment (MattToast) [inline smartdashboard/utils/StatusReader.py:181]: Oh! good catch!!

File: tests/test_StatusReader/test_format_status.py [hunk 2]
```diff
@@ -48,10 +48,17 @@
...
 def test_get_status(entity: t.Dict[str, t.Any], expected_status):
-    status_dir = entity.telemetry_metadata["status_dir"]
-    status = format_status(get_status(status_dir))
+    try:
+        status_dir = entity.telemetry_metadata["status_dir"]
+        status = format_status(get_status(status_dir))
+    except KeyError:
+        assert expected_status == f"Status: {StatusEnum.MALFORMED.value}"
+        return
     assert status == expected_status
```
Comment (MattToast) [inline tests/test_StatusReader/test_format_status.py:64]: Minor nit:
```suggestion
    try:
        status_dir = entity.telemetry_metadata["status_dir"]
    except KeyError:
        status = f"Status: {StatusEnum.MALFORMED.value}"
   else:
       status = format_status(get_status(status_dir))
    assert status == expected_status
```

Comment (MattToast) [reply tests/test_StatusReader/test_format_status.py:64]: Also, where is this `KeyError` being rasied from? Is it from the `entity.telemetry_metadata["status_dir"]`? If so, it doesn't seem like this route is testing any of the application code.

Comment (MattToast) [reply tests/test_StatusReader/test_format_status.py:64]: Per convo offline, it looks like the new parameters are not testing application code. Feel free to drop the new param and revert the `try/except` block

**Review by MattToast — CHANGES_REQUESTED** [review]
Couple of last minute change requests per offline convo, but otherwise LGTM!!

File: tests/test_StatusReader/test_format_status.py (hunk 2 above)
Comment (MattToast) [reply tests/test_StatusReader/test_format_status.py]: Resolved at https://github.com/CrayLabs/SmartDashboard/pull/41/commits/b1842491092060c1bfac148b67e19f79d705f952

**Review by MattToast — APPROVED** [review]
LGTM, pending tests!!
